src/inmanta/agent/agent.py

```diff
@@ -900,9 +899,8 @@ async def prepare_resources(
         loaded_resources: list[ResourceDetails] = []
         undeployable: dict[ResourceVersionIdStr, const.ResourceState] = {}
 
-        logs: dict[str, set[data.LogLine]] = defaultdict(set)
-        failed_resource_ids: dict[str, set[str]] = defaultdict(set)
-        logged_resource_types = set()
+        # {resource_type -> (set{resource_ids}, LogLine)}
```

Comment:
`typing.NewType` or plain type aliases can also be useful here, but I have no objections to this.